backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/CloneVmCommand.java

```diff
@@ -612,6 +612,10 @@ private void attachDetachDisk(Disk disk, ActionType actionType) {
     }
 
     private void setupParameters() {
+        if (getParameters().isEdited()) {
+            getParameters().setEditedVm(getParameters().getVm());
+        }
```

Comment:
it's so confusing now - the client is expected to provide the edited vm as parameters#vm and set the `edited` flag and then we move parameters#vm to parameters#editedVm and override parameters#vm with the vm from the database..

it's probably the safest way to do it though at this point so let's go with it - please add a comment to the parameters that `editedVm` is only supposed to be set internally so we won't get confused if/when we add this to the api

Reply:
That's how it worked before my changes - and that's because `CloneVmCommand` is inherited from `AddVmCommand` and `vm` parameter of `AddVmCommand` should be the VM to be created. The good and correct way to avoid this confusion is to make `CloneVmCommand` to _call_ `AddVmCommand` instead of _inheriting_ it.

I'll add a comment to `editedVm` field. But, generally, such confusion is common for all commands that use parameters for preserving their internal state.

Reply:
yes, there's a mixture of input arguments and internal state in the parameters but here it's more confusing - as there's 'vm', 'edited' and 'editedVm' in the parameters so it's harder to distinguish what should the client set and what is part of the internal state..

changing CloneVm so it won't extend AddVm is probably right but too risky - if you want to give it a try to avoid overriding the specified vm with the one from the database, it should be easier but I don't think it is worth the trouble, good documentation would do it